doc/user-guide/Push-notifications.md

```diff
@@ -0,0 +1,209 @@
+In order to allow clients to allocate such a node, we need to enable it in our [mod_pubsub][] on the MongooseIM server that will communicate with the _XMPP Push Service_.
+
+The minimal [mod_pubsub][]'s configuration looks as follows:
+
+```Erlang
+{mod_pubsub, [
+    {plugins, [<<"push">>]}}
+]}.
+```
+
+Such configuration will enable the [mod_pubsub][] with only one node type available: `push`.
+Please note that if you want use [mod_pubsub][] as a 'normal' publish-subscribe service, you just need to append the `<<"push">>` node type to the `plugins` list.
+Also, it's important to note, that the first node type on the `plugins` list, will be the default one (allocated when the client does not provide a node type in the node create stanza).
+
+##### [mod_push_service_mongoosepush][]
+
+This module acts as a bridge between [mod_pubsub][] that receives notifications from [mod_push][] and passes those to [MongoosePush][] which sends them to _FCM_ and/or _APNS_. To enable this module type in your configuration:
+
+```Erlang
+{http_connections, [{mongoose_push_http,
+    [{server, "https://localhost:8443"}]
+}]}.
+
+{mod_push_service_mongoosepush, [
+    {pool_name, mongoose_push_http}
+    {api_version, "v1"}
+]}.
+```
+
+First, we create the HTTP pool for communicating with [MongoosePush][].
+Here, we assume that [MongoosePush][] will be available on the localhost on port 8443 which is the default one.
+Next we enable [mod_push_service_mongoosepush][].
+First option is the name of the HTTP pool to use and the second one is the version of [MongoosePush][]'s API (currently only "_v1_" is supported).
+
+And that's it, we've just completed the entire MongooseIM configuration.
+All we need to do now is to set up [MongoosePush][].
+
+### Starting [MongoosePush][]
+
+The easiest way to start [MongoosePush][] is using its [docker image](https://hub.docker.com/r/mongooseim/mongoose-push).
+But before you can set [MongoosePush][] up, you need a _FCM_ application token and/or an _APNS_ application certificate.
+You can get the _FCM_ token [here](https://console.firebase.google.com/) and the easiest way of getting an _APNS_ application certificate is by running [this](https://github.com/fastlane/fastlane/tree/master/pem) script (please note that you need the certificate in `pem` format).
+
+After you get the _FCM_ application token and/or the _APNS_ application certificate, you can prepare to start [MongoosePush][].
+Firstly, prepare the following files structure:
+
+* priv/
+    * ssl/
+        * rest_cert.pem - The REST endpoint certificate
+        * rest_key.pem - private key for the REST endpoint certificate
+    * apns/
+        * prod_cert.pem - Production APNS app certificate
+        * prod_key.pem - Production APNS app certificate's private key
+        * dev_cert.pem - Development APNS app certificate
+        * dev_jey.pem - Development APNS app certificate's private key
```

Comment:
I've made a typo commenting on a typo :laughing: Of course I suggested `dev_key.pem` :)